tests/modules/commerce_redirect_test/src/Controller/RedirectTestController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Drupal\commerce_redirect_test\Controller;
+
+use Drupal\commerce\Response\NeedsRedirectException;
+
+/**
+ * Testing controller for redirect exception.
+ */
+class RedirectTestController {
+
+  /**
+   * Renders the page without a redirect exception.
+   *
+   * @return array
+   *   The render array.
+   */
+  public function noThrow() {
+    return [
+      '#markup' => 'Exception not thrown for secure redirect',
+    ];
+  }
+
+  /**
+   * Throws a local redirect response.
+   */
+  public function throwLocal() {
+    throw new NeedsRedirectException('/commerce_redirect_test/no_throw');
```

Comment:
We don't need to test controllers at all, they have their ways of redirecting. The point of this PR is to allow immediate redirects in nested forms.